freqtrade/vendor/qtpylib/indicators.py

```diff
@@ -103,47 +104,50 @@ def session(df, start='17:00', end='16:00'):
 
     return df.copy()
 
-
 # ---------------------------------------------
 
+
 def heikinashi(bars):
     bars = bars.copy()
     bars['ha_close'] = (bars['open'] + bars['high'] +
                         bars['low'] + bars['close']) / 4
 
-    bars['ha_open'] = (bars['open'].shift(1) + bars['close'].shift(1)) / 2
-    bars.loc[:1, 'ha_open'] = bars['open'].values[0]
-    for x in range(2):
-        bars.loc[1:, 'ha_open'] = (
-            (bars['ha_open'].shift(1) + bars['ha_close'].shift(1)) / 2)[1:]
+    # ha open
+    bars.loc[:1, 'ha_open'] = (bars['open'] + bars['close']) / 2
```

Comment:
The problem seems to be the :1 (if i remove it, tests pass).

we'll  need to dig into it as i don't think that's correct (i will try to talk to a pandas expert about this ...)

Reply:
I'm not sure about the correct formulae for heikinashi bars - but it did change here (result for open is different between old and new implementation). 

Also - the new implementation is a LOT slower ...

Reply:
How have you tested performance (mentioned above)?

Just issuing the backtesting command with def. strategy, but with a longer timerange?